lms/services/exceptions.py

```diff
@@ -258,6 +258,29 @@ def __init__(self, file_id):
         super().__init__(self.details)
 
 
+class SerializableError(Exception):
+    """An exception compatible with our default error handling for APIs."""
+
+    def __init__(
+        self,
+        message: str = None,
+        error_code: Optional[str] = None,
+        details: Optional[dict] = None,
+    ):
+        """
+        Initialise the error.
+
+        :param message: Message to display to the user (if any)
+        :param error_code: Error code to allow client side code to detect
+            this error
+        :param details: A JSON serializable payload of extra info
+        """
+        super().__init__(message)
+        self.error_code = error_code
+        self.message = message
+        self.details = details
```

Comment:
We have some default error handling for our API end-points but:

 * The mechanism is duck typed and kind of non-obvious
 * You have to create compatible classes on your own
 * It doesn't support a "message" to show the user

This class codifies the things you have to do to get this to work